File: Plugunit/Plugunit1/app/src/main/java/com/ysr/plugunit1/MainActivity.java
package com.ysr.plugunit1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.qihoo360.replugin.loader.a.PluginAppCompatActivity;

/**
 * 若插件Activity继承AppCompatActivity，需要在清单文件设置主题
 * <p>
 * Created by ysr on 2017/7/5 17:00.
 * 邮箱 ysr200808@163.com
 */
public class MainActivity extends PluginAppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button button = (Button) findViewById(R.id.btn);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                startActivity(new Intent(MainActivity.this, Main2Activity.class));
            }
        });
    }
}

File: BaseRePlugin/app/src/main/java/com/ysr/basereplugin/MainActivity.kt
package com.ysr.basereplugin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.qihoo360.replugin.RePlugin
import kotlinx.android.synthetic.main.activity_main.*

/**
 * 宿主App主页
 *
 * Created by ysr on 2017/7/04 20:24.
 * 邮箱 yfeng1023@gmail.com
 *
 * 加载内置App直接将App文件属性改成jar
 * 安装一个外置插件RePlugin.install("/sdcard/demo.apk");
 *  RePlugin.install也可以用于插件升级
 *  RePlugin.uninstall插件卸载
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn.setOnClickListener {//打开第一个app
            RePlugin.startActivity(this@MainActivity, RePlugin.createIntent("firstapp", "com.ysr.plugunit1.MainActivity"))
        }
        btnTwo.setOnClickListener {//打开第二个app
            RePlugin.startActivity(this@MainActivity, RePlugin.createIntent("twoapp", "com.ysr.plugunit2.MainActivity"))
        }
        butThree.setOnClickListener {//打开数据展示模块
            RePlugin.startActivity(this@MainActivity, RePlugin.createIntent("demo1", "com.qihoo360.replugin.sample.demo1.MainActivity"))
        }
    }
}
